### ISSUE TITLE
Dynamo Bug: fx graph gets wrong input shape information when having inplace view op

### ISSUE THREAD

**Issue by sherry-1001** [pytorch/pytorch#106693]
### 🐛 Describe the bug

fx graph gets wrong input shape information when having inplace view op.  The sample code is as follows:

```python
import numpy as np
np.random.seed(1335087076)
# device = "cpu"
tensor_0 = torch.from_numpy(np.random.uniform(0.0, 1.0, size=[3, 3, 2, 2])).to(torch.float32).requires_grad_(False)


def basic(tensor_0):
    # export XMLIR_XDNN_PYTORCH_CHECK_ENABLE_FALLBACK_BOOL=0
    tensor_3 = tensor_0.unsqueeze_(1)
    return tensor_3

dynamo_res = torch.compile(basic, backend="inductor")(tensor_0.clone())

cpu_res = basic(tensor_0.clone())

print("res: ", dynamo_res.shape, cpu_res.shape)
```

### Versions

PyTorch version: 2.0.1+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: 10.0.0-4ubuntu1 
CMake version: version 3.22.2
Libc version: glibc-2.31

Python version: 3.8.16 (default, Jun 12 2023, 18:09:05)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-4.14.0_1-0-0-50-x86_64-with-glibc2.17
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
   sorry, privacy 😊

Versions of relevant libraries:
[pip3] flake8==6.0.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.23.5
[pip3] torch==2.0.1+cpu
[pip3] torchtext==0.4.0
[pip3] torchvision==0.15.2+cpu
[conda] numpy                     1.23.5                   pypi_0    pypi
[conda] torch                     2.0.1+cpu                pypi_0    pypi
[conda] torchtext                 0.4.0                    pypi_0    pypi
[conda] torchvision               0.15.2+cpu               pypi_0    pypi

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305

**Comment by bdhirsh** [comment]
@sherry-1001 can you try again on a nightly? This bug may have been fixed since the 2.0.1 release. I just tried on a recent commit from the main branch, and I get this (ran your script as is, other than it was missing an `import torch` at the top):
```
res:  torch.Size([3, 1, 3, 2, 2]) torch.Size([3, 1, 3, 2, 2])
```

**Comment by sherry-1001** [comment]
> 

@bdhirsh Sorry, I forgot `import torch`. 
  I tried again on master (Source code compilation),  I also met this error. Then I check the source code:
`@register_meta(aten.unsqueeze_.default)
  def meta_unsqueeze_(self, dim):
    dim = maybe_wrap_dim(dim, self.dim() + 1)
    g_sizes, g_strides = inferUnsqueezeGeometry(self, dim)
    self.as_strided_(g_sizes, g_strides)
    return self`
here, self'shape has changed. 
Could you give me a commit id which you got correct result?
